### PR TITLE
[Expo Go] Prevent font mutation error

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

As explained in the comment:

```
Here, we extend a broken React Native function that has thread saftey issues,
if the app is reloaded multiple times, it has increased potential of crashing due to:
`Thread 163: "*** Collection <NSConcretePointerArray: 0x600003088000> was mutated while being enumerated."`
https://github.com/facebook/react-native/blob/1ecd98adc7b18504600fc7e9d56c7b4c9a090abd/Libraries/Text/RCTTextAttributes.m#L195
For now, we'll skip over the array's enumerator, and using a loop.
```

# How

# Test Plan

Spamming reload in the terminal won't cause the font error... it will however eventually crash on the next leak, this [unsafe call](https://github.com/facebook/react-native/blob/df331aeaa148cb04afb3b11fae4d45812b8ec06a/React/Base/RCTModuleData.mm#L534-L536) which invokes expo-updates before the scoped module is created, causing it to crash on Expo.plist missing.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-modules-core/ios/Services/EXReactFontManager.m [hunk 1]
```diff
@@ -24,7 +24,13 @@ + (UIFont *)EXUpdateFont:(UIFont *)uiFont
          scaleMultiplier:(CGFloat)scaleMultiplier
 {
   UIFont *font;
-  for (id<EXFontProcessorInterface> fontProcessor in currentFontProcessors) {
+  // Here, we extend a broken React Native function that has thread saftey issues,
+  // if the app is reloaded multiple times, it has increased potential of crashing due to:
+  // `Thread 163: "*** Collection <NSConcretePointerArray: 0x600003088000> was mutated while being enumerated."`
+  // https://github.com/facebook/react-native/blob/1ecd98adc7b18504600fc7e9d56c7b4c9a090abd/Libraries/Text/RCTTextAttributes.m#L195
+  // For now, we'll skip over the array's enumerator, and using a loop.
```
Comment (EvanBacon) [inline packages/expo-modules-core/ios/Services/EXReactFontManager.m:31]:
```suggestion
  // For now, we'll skip over the array's enumerator, and use a for-in loop.
```

File: packages/expo-modules-core/ios/Services/EXReactFontManager.m [hunk 2]
```diff
@@ -24,7 +24,13 @@ + (UIFont *)EXUpdateFont:(UIFont *)uiFont
          scaleMultiplier:(CGFloat)scaleMultiplier
 {
   UIFont *font;
-  for (id<EXFontProcessorInterface> fontProcessor in currentFontProcessors) {
+  // Here, we extend a broken React Native function that has thread saftey issues,
```
Comment (ide) [inline packages/expo-modules-core/ios/Services/EXReactFontManager.m:27]:
```suggestion
  // Here, we extend a broken React Native function that has thread safety issues,
```

File: packages/expo-modules-core/ios/Services/EXReactFontManager.m [hunk 3]
```diff
@@ -24,7 +24,13 @@ + (UIFont *)EXUpdateFont:(UIFont *)uiFont
...
+  // if the app is reloaded multiple times, it has increased potential of crashing due to:
+  // `Thread 163: "*** Collection <NSConcretePointerArray: 0x600003088000> was mutated while being enumerated."`
+  // https://github.com/facebook/react-native/blob/1ecd98adc7b18504600fc7e9d56c7b4c9a090abd/Libraries/Text/RCTTextAttributes.m#L195
+  // For now, we'll skip over the array's enumerator, and use a for-in loop.
```
Comment (ide) [inline packages/expo-modules-core/ios/Services/EXReactFontManager.m:31]:
```suggestion
  // For now, we'll skip over the array's enumerator, and use an indexed for-loop.
```

**Review by ide — CHANGES_REQUESTED** [review]
Reading the code, this looks potentially like a bug in our code and not RN's. We define `currentFontProcessors` and presumably RN calls into functions that access this array.

What threads are used to access this array?

I would like to understand why this array is accessed concurrently -- if it is not intentional to do so, we should fix that. If it is intentional, a lock would be better. Using a for-loop to index into the array covers up the issue but does not fix it. The concurrent array access is still a problem.

**Pull request closed without merging**